doc/design/overview.md

```diff
@@ -0,0 +1,90 @@
+# Overview of Astacus design #
+
+While more thorough documentation would be welcome, time constraints are
+what they are, so here's very brief overview of the Astacus design.
+
+## Components
+
+Astacus is designed so that every cluster node runs it. Internally, it has
+two different parts:
+
+- `astacus.coordinator`, which is responsible for using set of Astacus
+  'node' components to perform user-requested operation. The REST endpoints
+  it provides are mostly asynchronous.
+
+- `astacus.node`, which is responsible for performing (parts) of tasks
+  required by users that apply to the current operation being performed. A
+  lot of it is implemented as synchronus REST endpoints to be used
+  internally by the coordinator.
+
+
+Additionally, the code also has `astacus.common` which contains some code
+that is shared by the both parts of the code.
```

Comment:
Maybe re-phrase this a bit... ", the project also contains `astacus.common` which contains code shared by coordinator and node." ... or something